vault/resource_ldap_secret_backend_test.go

```diff
@@ -62,51 +62,132 @@ func TestLDAPSecretBackend(t *testing.T) {
+//
+// export AD_URL=ldaps://localhost:2636
+func TestLDAPSecretBackend_SchemaAD(t *testing.T) {
+	var (
+		path         = acctest.RandomWithPrefix("tf-test-ldap")
+		resourceType = "vault_ldap_secret_backend"
+		resourceName = resourceType + ".test"
+		userDN       = "CN=Users,DC=corp,DC=example,DC=net"
+		bindPass     = "SuperSecretPassw0rd"
+		bindDN       = "CN=Administrator,CN=Users,DC=corp,DC=example,DC=net"
+
+		url = testutil.SkipTestEnvUnset(t, "AD_URL")[0]
+	)
+	resource.Test(t, resource.TestCase{
+		ProviderFactories: providerFactories,
+		PreCheck: func() {
+			testutil.TestAccPreCheck(t)
+			SkipIfAPIVersionLT(t, testProvider.Meta(), provider.VaultVersion112)
+		},
+		CheckDestroy: testCheckMountDestroyed(resourceType, consts.MountTypeLDAP, consts.FieldPath),
+		Steps: []resource.TestStep{
 			{
-				Config: testLDAPSecretBackendConfig(path, updatedDescription, bindDN, bindPass, url, updatedUserDN, "ad", false),
+				Config: testLDAPSecretBackendConfig_ad(path, url, ""),
 				Check: resource.ComposeTestCheckFunc(
-					resource.TestCheckResourceAttr(resourceName, consts.FieldPath, path),
+					func(*terraform.State) error {
+						client := testProvider.Meta().(*provider.ProviderMeta).GetClient()
+						if _, err := client.Logical().Write(path+"/rotate-root", nil); err != nil {
+							return err
+						}
+						return nil
+					},
 					resource.TestCheckResourceAttr(resourceName, consts.FieldSchema, "ad"),
-					resource.TestCheckResourceAttr(resourceName, consts.FieldDescription, updatedDescription),
-					resource.TestCheckResourceAttr(resourceName, consts.FieldDefaultLeaseTTL, "3600"),
-					resource.TestCheckResourceAttr(resourceName, consts.FieldMaxLeaseTTL, "7200"),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldPath, path),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldDescription, ""),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldDefaultLeaseTTL, "0"),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldMaxLeaseTTL, "0"),
 					resource.TestCheckResourceAttr(resourceName, consts.FieldBindDN, bindDN),
 					resource.TestCheckResourceAttr(resourceName, consts.FieldBindPass, bindPass),
 					resource.TestCheckResourceAttr(resourceName, consts.FieldURL, url),
-					resource.TestCheckResourceAttr(resourceName, consts.FieldUserDN, updatedUserDN),
-					resource.TestCheckResourceAttr(resourceName, consts.FieldInsecureTLS, "false"),
-					resource.TestCheckResourceAttr(resourceName, consts.FieldConnectionTimeout, "99"),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldUserDN, userDN),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldInsecureTLS, "true"),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldConnectionTimeout, "30"),
+				),
+			},
+			{
+				Config: testLDAPSecretBackendConfig_ad(path, url, `description = "new description"`),
+				Check: resource.ComposeTestCheckFunc(
+					func(*terraform.State) error {
```

Comment:
I think what we have is fine, but I wonder if moving the rotate root into a `PreConfig` function would fit more with patterns already in the codebase. Also it looks like we're rotating the root twice, is there a particular reason?

Reply:
I considered using PreConfig but in this case I actually don't want it to run before the config is applied. In this TestStep I am first updating the `description` in the TF config to replicate the events that triggered the issue of overwriting `bindpass` and then calling rotate-root.

I could update the `description` in one step and then call PreConfig with rotate-root in the next step. And I considered that too but I didn't feel too strongly about changing it. Now that you have commented on it I think you have persuaded me to change it since having this call within a `Check` may be confusing for people.

Reply:
I am going to punt this to a future improvement